pipeline_dp/dataset_histograms/computing_histograms.py

```diff
@@ -229,6 +293,57 @@ def _compute_linf_contributions_histogram(
                                         hist.HistogramType.LINF_CONTRIBUTIONS)
 
 
+def _compute_linf_sum_contributions_histogram(
+        col, backend: pipeline_backend.PipelineBackend):
+    """Computes histogram of per partition privacy id contributions.
+
+    This histogram contains: the number of (privacy id, partition_key)-pairs
+    which have sum of values X_1, X_2, ..., X_n, where X_1 = min_sum,
+    X_n = one before max sum and n is equal to
+    NUMBER_OF_BUCKETS_IN_LINF_SUM_CONTRIBUTIONS_HISTOGRAM.
+
+    Args:
+        col: collection with elements (privacy_id, partition_key, value).
+        backend: PipelineBackend to run operations on the collection.
+    Returns:
+        1 element collection, which contains the computed hist.Histogram.
+    """
+    col = backend.sum_per_key(
+        col, "Sum of contributions per (privacy_id, partition)")
+    # col: ((pid, pk), sum_per_key)
+    col = backend.values(col, "Drop keys")
+    # col: (float)
+    col = backend.to_multi_transformable_collection(col)
+    lowers = _min_max_lowers(
+        col, NUMBER_OF_BUCKETS_IN_LINF_SUM_CONTRIBUTIONS_HISTOGRAM, backend)
+
+    return _compute_frequency_histogram_helper_with_lowers(
+        backend, col, hist.HistogramType.LINF_SUM_CONTRIBUTIONS, lowers)
+
+
+def _min_max_lowers(col, number_of_buckets,
+                    backend: pipeline_backend.PipelineBackend):
+    """Returns bin lowers equally distributed between min and max elements of
+    the collection.
+
+    The method determines min and max values of the given collection and
+    returns bin lowers that equally divide the range in `number_of_buckets`.
+
+    Args:
+        col: collection of numbers.
+        backend: PipelineBackend to run operations on the collection.
+    Returns:
+        collection of lowers (float,).
+    """
+    min_max_values = pipeline_functions.min_max_elements(
+        backend, col, "Min and max value in dataset")
+    return backend.flat_map(
+        min_max_values,
+        lambda min_max: numpy.linspace(min_max[0], min_max[1],
+                                       (number_of_buckets + 1))[:-1],
+        "flatMap to lowers")
```

Comment:
Isn't it better to return a collection with 1 element  - np.array with lower? In that case we have guaranetees on the order of the elements (we need guarantees to run binary search)